api/prometheus/v1/api.go

```diff
@@ -1138,6 +1138,9 @@ func (h *apiClientImpl) DoGetFallback(ctx context.Context, u *url.URL, args url.
 		return nil, nil, nil, err
 	}
 	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+	// Underlying `net.http` library automatically retries` idempotent requests when connectivity issues are hit.
+	// POST requests are not considered idempotent by default, so we need to explicitly mark them as such.
+	req.Header["Idempotency-Key"] = nil
```

Comment:
Could you please at the docs directly here?
```
Transport only retries a request upon encountering a network error if the request is idempotent and either has no body or has its Request.GetBody defined. HTTP requests are considered idempotent if they have HTTP methods GET, HEAD, OPTIONS, or TRACE; or if their Header map contains an "Idempotency-Key" or "X-Idempotency-Key" entry. If the idempotency key value is a zero-length slice, the request is treated as idempotent but the header is not sent on the wire.
```

And a link to the docs?

Reply:
PTAL.

I think we should squash before merging. Not sure if I am suppose to do that, or the Github's UI is used for it?